tests/contrib/botocore/test.py

```diff
@@ -4247,3 +4247,107 @@ def test_aws_payload_tagging_kinesis(self):
             with self.tracer.trace("kinesis.manual_span"):
                 client.create_stream(StreamName=stream_name, ShardCount=1)
                 client.put_records(StreamName=stream_name, Records=data)
+
+    # Peer service tests
+    @mock_sqs
+    @TracerTestCase.run_in_subprocess(env_overrides=dict(AWS_LAMBDA_FUNCTION_NAME="my-func"))
+    def test_sqs_client_peer_service_in_lambda(self):
+        """Test that peer.service tag is set for SQS when running in AWS Lambda"""
+        sqs = self.session.create_client("sqs", region_name="us-east-1")
+        pin = Pin(service=self.TEST_SERVICE)
+        pin._tracer = self.tracer
+        pin.onto(sqs)
+
+        sqs.list_queues()
+        spans = self.get_spans()
+        assert spans
+        assert len(spans) == 1
+        span = spans[0]
+        # Should have peer.service set to sqs hostname
+        assert span.get_tag("peer.service") == "sqs.us-east-1.amazonaws.com"
+
+    @mock_s3
+    @TracerTestCase.run_in_subprocess(env_overrides=dict(AWS_LAMBDA_FUNCTION_NAME="my-func"))
+    def test_s3_client_peer_service_in_lambda(self):
+        """Test that peer.service tag is set for S3 when running in AWS Lambda"""
+        s3 = self.session.create_client("s3", region_name="us-east-1")
+        pin = Pin(service=self.TEST_SERVICE)
+        pin._tracer = self.tracer
+        pin.onto(s3)
+
+        # Test with bucket parameter
+        s3.create_bucket(Bucket="test-bucket")
+        spans = self.get_spans()
+        assert spans
+        assert len(spans) == 1
+        span = spans[0]
+        # Should have peer.service set to bucket-specific hostname
+        assert span.get_tag("peer.service") == "test-bucket.s3.us-east-1.amazonaws.com"
```

Comment:
This seems like a good candidate for a snapshot test since it's making assertions about the number and content of spans

Reply:
The number of spans is irrelevant, we are just mainly checking the existence of the correct peer.service tags. Do we really need snapshots for one tag check?

Reply:
It's not strictly necessary in my opinion

Reply:
In that case lets forgo it